src/index.ts

```diff
@@ -23,45 +25,65 @@ const EXECUTION_COMPLETED_STATUSES = [
 ];
 const GITHUB_BASE_URL = 'https://api.github.com';
 
-async function run(): Promise<void> {
+export function optionalArrayInput(name: string): string[] {
+  // Note: GitHub Action inputs default to '' for undefined inputs, remove these
+  return core
+    .getInput(name, {
+      required: false,
+    })
+    .split(/[,\n]/)
+    .filter((item) => item.length)
+    .map((item) => item.trim());
+}
+
+export function optionalInput(name: string): string | undefined {
+  const rawValue = core.getInput(name, {
+    required: false,
+  });
+
+  if (rawValue.length > 0) {
+    return rawValue;
+  }
+  return;
+}
+
+export function booleanInput(name: string): boolean {
+  return (
+    core
+      .getInput(name, {
+        required: false,
+      })
+      .toLowerCase() === 'true'
+  );
+}
+
+export async function run(): Promise<void> {
   try {
     core.startGroup('Gathering inputs');
-    const applicationId: string = core.getInput('application-id', {
-      required: false,
-    });
-    const environmentId: string = core.getInput('environment-id', {
-      required: false,
-    });
+    const applicationId = optionalInput(ActionInputs.ApplicationId);
+    const environmentId = optionalInput(ActionInputs.EnvironmentId);
 
-    const apiKey: string = process.env.MABL_API_KEY || '';
+    const apiKey = process.env.MABL_API_KEY;
```

Comment:
We should bail if no key is defined, rather than wait for API call to auth error.